src/raftstore/store/store.rs

```diff
@@ -1813,32 +1828,88 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         epoch: metapb::RegionEpoch,
         split_key: Vec<u8>,
     ) {
+        self.on_split_region(split_key, Some(region_id), Some(epoch), None);
+    }
+
+    fn on_split_region(
+        &mut self,
+        split_key: Vec<u8>,
+        region_id: Option<u64>,
+        epoch: Option<metapb::RegionEpoch>,
+        cb: Option<Callback>,
+    ) {
+        let region_id = match region_id {
+            Some(id) => id,
+            None => match self.region_ranges
+                .range((Excluded(split_key.clone()), Unbounded::<Key>))
+                .next()
```

Comment:
Need to check if the region contains `split_key`?
btw, is it ok if `split_key == region.start_key`?

Reply:
seem we still need to check the split key validation like the split coprocessor does.